src/main/java/io/kestra/plugin/microsoft365/sharepoint/Create.java

```diff
@@ -0,0 +1,182 @@
+        Logger logger = runContext.logger();
+
+        String rParentId = runContext.render(parentId).as(String.class).orElseThrow();
+        String rName = runContext.render(name).as(String.class).orElseThrow();
+        String rContent = content != null ? runContext.render(content).as(String.class).orElse(null) : null;
+
+        ItemType rItemType = runContext.render(itemType).as(ItemType.class).orElse(ItemType.FILE);
+
+        SharepointConnection connection = this.connection(runContext);
+        GraphServiceClient client = connection.createClient(runContext);
+        String driveId = connection.getDriveId(runContext, client);
+        String siteId = connection.getSiteId(runContext);
+
+        DriveItem createdItem;
+
+        if (rItemType == ItemType.FOLDER) {
+            DriveItem newFolder = new DriveItem();
+            newFolder.setName(rName);
+            newFolder.setFolder(new Folder());
+
+            createdItem = client.drives().byDriveId(driveId)
+                .items().byDriveItemId(rParentId)
+                .children()
+                .post(newFolder);
+            logger.info("Created folder '{}' in parent '{}'", rName, rParentId);
+        } else {
+            // Create a file using simple upload
+            // For small files, use PUT to /drives/{drive-id}/items/{parent-id}:/{filename}:/content
+            byte[] contentBytes = (rContent != null) ? rContent.getBytes() : new byte[0];
+
+            java.io.ByteArrayInputStream inputStream = new java.io.ByteArrayInputStream(contentBytes);
+
+            createdItem = client.drives().byDriveId(driveId)
+                .items().byDriveItemId(rParentId + ":/" + rName + ":")
+                .content()
+                .put(inputStream);
+            logger.info("Created file '{}' in parent '{}'", rName, rParentId);
+        }
+
+        return Output.builder()
+            .itemId(createdItem.getId())
+            .itemName(createdItem.getName())
+            .webUrl(createdItem.getWebUrl())
+            .isFolder(createdItem.getFolder() != null)
+            .isFile(createdItem.getFile() != null)
+            .build();
+    }
+
+    public enum ItemType {
+        FILE,
+        FOLDER
+    }
+
+    @Builder
+    @Getter
+    public static class Output implements io.kestra.core.models.tasks.Output {
+        @Schema(
+            title = "The ID of the created item."
+        )
+        private final String itemId;
+
+        @Schema(
+            title = "The name of the created item."
+        )
+        private final String itemName;
+
+        @Schema(
+            title = "The web URL of the created item."
+        )
+        private final String webUrl;
+
+        @Schema(
+            title = "Whether the created item is a folder."
+        )
+        private final Boolean isFolder;
+
+        @Schema(
+            title = "Whether the created item is a file."
+        )
+        private final Boolean isFile;
```

Comment:
Actually we already have the info as an input property so no need to repeat it in the output